lib/internal/http2/compat.js

```diff
@@ -97,43 +97,45 @@ function onStreamError(error) {
 }
 
 function onRequestPause() {
-  this[kStream].pause();
+  const stream = this[kStream];
+  if (stream)
+    stream.pause();
 }
 
 function onRequestResume() {
-  this[kStream].resume();
-}
-
-function onRequestDrain() {
-  if (this.isPaused())
-    this.resume();
+  const stream = this[kStream];
+  if (stream)
+    stream.resume();
 }
 
-function onStreamResponseDrain() {
+function onStreamDrain() {
   this[kResponse].emit('drain');
 }
 
+// TODO Http2Stream does not emit 'close'
 function onStreamClosedRequest() {
   this[kRequest].push(null);
 }
 
+// TODO Http2Stream does not emit 'close'
```

Comment:
I can remove these comments but I wanted it clear that these two don't run as things are. It's a bit confusing otherwise.

Reply:
@jasnell can we remove those then? Or do you prefer if we add 'close' to Http2Stream?

Reply:
I'm good with removing the comments.

Reply:
oh lol... heh Matteo pointed out that I didn't understand the comments :-) This is true, I missed that.

I'm thinking we need to refactor this so that `streamClosed` and `close` are one and the same and that `close` is emitted appropriately. The one bit we'd have to account for is the `code` that is reported in `streamClosed`

Reply:
I could be wrong but I think @mcollina is asking about removing the functions which the comments refer to. If a 'close' event will be added to Http2Stream at some point then they should be kept but if that's not going to happen then we can safely remove.